Automattic/jetpack
# React compatibility: Removal of react-test-renderer

Fixes https://github.com/Automattic/jetpack/issues/38753

## Proposed changes:

* We are no longer using `react-test-renderer` in any plugin / package except the connection js-package, so it has been removed. This package will be deprecated in React 19 and we would have needed a replacement otherwise.
* In the `js-packages/connection` package, instead of replacing the `react-test-render/shallow` usage with `react-shallow-renderer`, we'll just import render from '@testing-library/react'; which is preferred instead of using `react-test-renderer` itself, and works well in the tests. The package `react-test-render/shallow` would be removed in React 19.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/38753

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* To confirm the removal doesn't cause an issue for tests, you can run the following tests locally:
* 'jetpack test', select 'plugins', 'Jetpack', and then 'js'.
* Follow the same approach for `plugins/crm`, `packages/videopress`, `packages/my-jetpack`, `js-packages/partner-coupon`, `js-packages/licensing`, `js-packages/idc`, `js-packages/components` (all js tests).
